src/main/java/nl/tudelft/semgroup4/OptionsState.java

```diff
@@ -26,95 +23,236 @@
-            }
+    /**
+     * Poll the sound button.
+     */
+    private void soundUpdate() {
+        if (soundOnOff.isMouseOver() && input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
+            Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
+                    "User toggled sound " + (ResourcesWrapper.musicOn ? "off" : "on") );
+            controller.toggleMusic();
         }
-        if (backButton.isMouseOver()) {
-            if (input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
-                Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
-                        "User goes back to main menu" );
-                game.enterState(States.StartScreenState);
-            }
+    }
+
+    /**
+     * Poll the back button.
+     * @param game
+     *             the game for which the state can be changed.
+     */
+    private void backUpdate(StateBasedGame game) {
+        if (backButton.isMouseOver() && input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
+            Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
+                    "User goes back to main menu" );
+            game.enterState(States.StartScreenState);
         }
-        if (loggerButton.isMouseOver()) {
-            if (input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
-                Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
-                        "User goes to logger settings");
-                toggleLoggerSet();
-            }
+    }
+
+    /**
+     * Poll the loger button.
+     */
+    private void loggerUpdate() {
+        if (loggerButton.isMouseOver() && input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
+            Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
+                    "User goes to logger settings");
+            toggleLoggerSet();
         }
-        if (keyButton.isMouseOver()) {
-            if (input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
-                Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
-                        "User goes to key binding settings");
-                game.enterState(States.KeyBindState);
-            }
+    }
+
+    /**
+     * Poll the KeyBindings button.
+     * @param game
+     *             the game for which the state can be changed.
+     */
+    private void keyUpdate(StateBasedGame game) {
+        if (keyButton.isMouseOver() && input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
+            Game.LOGGER.log(LogSeverity.DEBUG, "OptionsMenu",
+                    "User goes to key binding settings");
+            game.enterState(States.KeyBindState);
         }
     }
 
+    /**
+     * Toggle the logger screen enabled boolean.
+     * Handled by the controller.
+     */
     public void toggleLoggerSet() {
-        loggerSetEnabled = !loggerSetEnabled;
+        controller.toggleLoggerScreen();
     }
 
-
-
     @Override
     public int getID() {
-        return States.OptionsState;
+        return controller.getId();
+    }
+
+    protected boolean getLoggerSetEnabled() {
```

Comment:
*Auto-generated comment*

    
    A getX() method which returns a boolean should be named isX()